Sources/ProcedureKit/Procedure.swift

```diff
@@ -212,27 +286,28 @@ open class Procedure: Operation, ProcedureProtocol {
     final public var log: LoggerProtocol {
         get {
             let operationName = self.operationName
-            return protectedProperties.read { LoggerContext(parent: $0.log, operationName: operationName) }
+            return _stateLock.withCriticalScope { LoggerContext(parent: protectedProperties!.log, operationName: operationName) }
```

Comment:
Perhaps make `protectedProperties` an IUO, so you can avoid the `!` everywhere and still be able to set it to `nil`?